lib/govuk_index/presenters/indexable_content_presenter.rb

```diff
@@ -0,0 +1,31 @@
+module GovukIndex
+  class IndexableContentPresenter
+    def initialize(details)
+      @details = details
+      @sanitiser = IndexableContentSanitiser.new
+    end
+
+    def indexable_content
+      return nil if details.nil?
+      sanitiser.clean(elements)
+    end
+
+  private
+
+    attr_reader :details, :sanitiser
+
+    def elements
+      if details['parts']
+        parts(details)
+      else
+        details.values
+      end
+    end
```

Comment:
Just thinking if we want to support parts and anything else at the same time the following alternative might be nicer:
```
def elements
   details.map do |key, value|
      key == 'parts' ? parts(value) : value
  end
end
```